Fixes #3947 Adds Fake driver and fixes fluent tests (iteration-zero)

tznind opened the pull request:
Removes iteration timer on fluent tests to reveal multi threading issues and fixes them

- Tests should no longer fail randomly
- Should complete faster (~8s for 94 tests)
- When one test fails it should no longer bring down all future tests! (_coordinator was unexpectedly null)

Note this is super set of #4222 

## Fixes

- Fixes #3947

## Proposed Changes/Todos


Support creating `Application` and `Driver` level tests for v2.

- Driver level tests are those that use `[SetupFakeDriver]`
- Application level tests are those that use `[AutoInitShutdown]`

These tests need:
- To 'work' 
- To be able to resize 

### Implementation
To enable this workflow we first need to pull all the Dependency Injection elements out into a shared provider factory.  This lets us create classes but swap out Mocks and fakes as needed.

- Add IComponentFactory for creating system internal classes
  - Input
  - Output
  - WindowSize
  - etc

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [x] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

tznind commented:
Something is trying to load windows kernel32 in linux, not sure what

BDisp commented:
> Something is trying to load windows kernel32 in linux, not sure what

Some method is calling Win32 API in Linux. If it's calling inside a try/catch then it's only reporting that and doesn't fail in the unit tests, but if it isn't calling inside a try/catch then the unit test will fail.

tig commented on Examples/UICatalog/Scenarios/SendKeys.cs:
```diff
@@ -39,7 +39,7 @@ public override void Main ()
         txtResult.KeyDown += (s, e) =>
                              {
-                                 rKeys += (char)e.KeyCode;
+                                 rKeys += new Rune ((uint)(e.KeyCode & ~KeyCode.AltMask & ~KeyCode.CtrlMask & ~KeyCode.ShiftMask));
```
Please use `Key`, not `KeyCode`. `Key` has helpers for this. Since this is an example, we don't want to be encouraging devs to use `KeyCode`.

tznind replied:
Done in [14c68e5](https://github.com/gui-cs/Terminal.Gui/pull/4225/commits/14c68e550f04bce301b1e297d93e65ea1fc76070)

tig commented on Terminal.Gui/Drivers/V2/InputProcessor.cs:
```diff
@@ -179,6 +180,22 @@ internal bool IsValidInput (Key key, out Key result)
+
+            if (key.IsAlt)
+            {
+                result = new (result.KeyCode | KeyCode.AltMask);
```
`result = result.WithAlt`

tznind replied:
Done in [14c68e5](https://github.com/gui-cs/Terminal.Gui/pull/4225/commits/14c68e550f04bce301b1e297d93e65ea1fc76070)

tznind commented:
@tig have addressed feedback and fixed the warnings from xmldoc and also put each class in its own file (tidyup).

And always green ticks from CI 💪 - no more random failures
